lib/Dialect/FIRRTL/Export/FIREmitter.cpp

```diff
@@ -531,22 +537,56 @@ void Emitter::emitStatement(MemOp op) {
     indent() << "readwriter => " << readwriter << "\n";
 
   indent() << "read-under-write => ";
-  switch (op.ruw()) {
-  case RUWAttr::Undefined:
-    os << "undefined";
-    break;
-  case RUWAttr::Old:
-    os << "old";
-    break;
-  case RUWAttr::New:
-    os << "new";
-    break;
-  }
+  emitAttribute(op.ruw());
   os << "\n";
 
   reduceIndent();
 }
 
+void Emitter::emitStatement(SeqMemOp op) {
+  indent() << "smem " << op.name() << " : ";
+  emitType(op.getType());
+  os << " ";
+  emitAttribute(op.ruw());
+  emitLocationAndNewLine(op);
+}
+
+void Emitter::emitStatement(CombMemOp op) {
+  indent() << "cmem " << op.name() << " : ";
+  emitType(op.getType());
+  emitLocationAndNewLine(op);
+}
+
+void Emitter::emitStatement(MemoryPortOp op) {
+  // Nothing to output for this operation.
+}
+
+void Emitter::emitStatement(MemoryPortAccessOp op) {
+  indent();
+
+  // Print the port direction and name.
+  auto port = cast<MemoryPortOp>(op.port().getDefiningOp());
+  emitAttribute(port.direction());
+  os << " mport " << port.name() << " = ";
```

Comment:
If the port can later be used in an expression, you'll have to call `addValueName` here such that uses of the port's SSA value in an expression know what name to emit. Not sure how exactly the smem/cmem work, but I'd imagine that after `infer mport port0 = ...` you can do stuff like `foo <= port0` or `port0 <= bar`.

Reply:
Arg, good thing you spotted that mistake.  I added it above in the handling for `MemoryPortOp`, and added a test that just connects the `seqmem` and `combmem` ports together.